src/enclave/enclave_init.c

```diff
@@ -107,6 +117,26 @@ static int startmain(void* args)
     init_wireguard();
     find_and_mount_disks();
 
+    /* Generate TLS certificate and private key and save them to files.
+     * Once we have kernel/user separation, pass the cert and private
+     * key to user space through the stack, and move
+     * sgxlkl_write_tls_credentials to user space.
+     */
+
+    if (!sgxlkl_in_sw_debug_mode())
+    {
+        enclave_generate_tls_credentials(
```

Comment:
I think there's a layering problem here; we cant call `oe_*` at this point in time anymore. If I'm not mistaken, the consensus was that we would generate this kind of data earlier and then pass it to the application via auxv. This would also mean that applications could pick it up directly from memory instead of a file. The current implementation would also fail on a read-only root file system. (CC @davidchisnall)

Reply:
Yes, this routine is a part of the nominal userspace and will be refactored soon to make that distinction more explicit.  We may not call OE APIs from here.

More generally, given that we've had so many meetings discussing the design and the mechanism for passing this via the ELF aux vector, I'd like to understand why a different approach is being pursued here.